test/search/fragmentation_scheme_test.exs

```diff
@@ -0,0 +1,83 @@
+defmodule Search.FragmentationSchemeTest do
+  alias Search.FragmentationScheme
+  use ExUnit.Case, async: true
+
+  describe "split/2" do
+    test "when given an empty string, returns empty list" do
+      assert FragmentationScheme.split("") == []
+    end
+
+    test "when given a string which satisfies the size constraint, returns a singleton list with that string" do
+      str = "short string"
+
+      assert FragmentationScheme.split(str, max_size: 100) == [str]
+    end
+
+    test "when given a string that is too long and splitting along whitespace is possible, splits the string" do
+      str = "some words and some more words"
+
+      assert FragmentationScheme.split(str, max_size: 15) == [
+               "some words and ",
+               "some more words"
+             ]
+    end
+
+    test "when splitting along whitespace, respects non-space whitespace characters" do
+      str = "word\nword\tword\u{2003}word"
+
+      assert FragmentationScheme.split(str, max_size: 7) == [
+               "word\n",
+               "word\t",
+               "word\u{2003}",
+               "word"
+             ]
+    end
+
+    test "when splitting along whitespace and the text starts with whitespace, the whitespace characters are prepended to the first fragment" do
+      str = "    words and some more words"
+
+      assert FragmentationScheme.split(str, max_size: 15) == [
+               "    words and ",
```

Comment:
out of curiosity what's the rationale behind keeping trailing \n\t, leading whitespace, etc on returned words in the list?

Reply:
It's so that the original docs can be recreated from the fragments by simple concatenation. If further preprocessing that would discard some characters is needed, it can always be done on the fly just before embedding

Reply:
Got it, thanks!